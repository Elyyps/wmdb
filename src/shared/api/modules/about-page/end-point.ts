import { generateAboutPageDummyData } from "./generate-dummy-data";

export const generateAboutPageData = generateAboutPageDummyData;
