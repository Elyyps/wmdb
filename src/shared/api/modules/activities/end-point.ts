import { generateActivitiesDummyData } from "./dummy-data";

export const generateActivitiesData = generateActivitiesDummyData;
