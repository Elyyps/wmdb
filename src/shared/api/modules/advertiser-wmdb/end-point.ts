import { generateAdvertiserDummyData } from "./dummy-data";
export const generateAdvertiserData = generateAdvertiserDummyData;
