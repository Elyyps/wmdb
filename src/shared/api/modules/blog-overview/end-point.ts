import { generateBlogOverviewDummyData, generateCards } from "./dummy-data";
import { ICompanyOutingCard } from "@app/api/core/company-outing-card";

export const generateBlogOverviewData = generateBlogOverviewDummyData;
const amountOfCards = 50;
export const getCardsWithPagination = (skip: number, take: number): ICompanyOutingCard[] => {
  let cards: ICompanyOutingCard[] = generateCards(amountOfCards);
  cards = cards.slice(skip, take + skip);

  return cards;
};
