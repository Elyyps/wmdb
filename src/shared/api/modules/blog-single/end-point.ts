import { generateBlogSingleDummyData } from "./dummy-data";

export const generateBlogSingleData = generateBlogSingleDummyData;
