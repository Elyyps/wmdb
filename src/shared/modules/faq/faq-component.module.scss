.faq {
  &__columns {
    @media (max-width: $breakpoint-xsmall-max) {
      > div {
        margin-bottom: 16px;
      }
    }
    &__content {
      width: 350px;
    }
    ul {
      margin-top: 10px;
      margin-left: 25px;

      // li {
      //   list-style-type: none;
      // }
      li {
        list-style-type: disc;
      }
    }
  }
}
