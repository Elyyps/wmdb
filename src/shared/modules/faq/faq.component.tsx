import * as React from "react";
import styles from "./faq-component.module.scss";
import { Section } from "@app/core/section";
import { ExpendableComponent } from "@app/core/expendable";
import { IFAQ } from "@app/api/modules/faq/faq";
import { FAQFormComponent } from "@app/core/faq-form";

export interface IFAQComponentProps {
  faq: IFAQ;
}

const FAQComponent = (props: IFAQComponentProps) => (
  <Section paddingBottom={"72px"} paddingTop={"72px"}>
    <div className={` ${styles["faq__columns"]} uk-grid uk-child-width-1-2@s uk-grid-xlarge `}>
      <div>
        <h2>{props.faq.title}</h2>
        <ExpendableComponent modify="show-desktop" items={props.faq.items} />
      </div>
      <div>
        <h2>{props.faq.formTitle}</h2>
        <p>{props.faq.formContent}</p>
        <FAQFormComponent onSubmit={() => alert("submitted")} />
      </div>
    </div>
  </Section>
);

export { FAQComponent };
