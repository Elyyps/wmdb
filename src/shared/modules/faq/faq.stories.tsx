import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { FAQComponent } from "./faq.component";
import { generateFAQData } from "@app/api/modules/faq/end-point";

storiesOf("FAQ", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <FAQComponent faq={generateFAQData()} />);
