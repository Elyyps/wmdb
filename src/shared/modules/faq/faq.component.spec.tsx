import React from "react";
import { shallow } from "enzyme";
import { FAQComponent } from "./faq.component";
import { generateFAQData } from "@app/api/modules/faq/end-point";

describe("[FAQ]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<FAQComponent faq={generateFAQData()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
