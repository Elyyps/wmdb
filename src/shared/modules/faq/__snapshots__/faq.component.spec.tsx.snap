// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[FAQ] should render component without crashing 1`] = `
<Section
  paddingBottom="72px"
  paddingTop="72px"
>
  <div
    className=" faq__columns uk-grid uk-child-width-1-2@s uk-grid-xlarge "
  >
    <div>
      <h2>
        Veelgestelde vragen
      </h2>
      <ExpendableComponent
        items={
          Array [
            Object {
              "content": "Daarom is de prijs niet zomaar te noemen zonder uw informatie. Alle aanbieders maken voor u graag een offerte op maat. Indien u het aanvraagformulier onder het gekozen uitje invult, met uw wensen en aanvullingen, maken wij een offerte op maat voor u. Na het invullen ontvangt u ook direct het telefoonnummer van de organisator, zodat u indien gewenst ook direct telefonisch contact kunt opnemen.",
              "title": "Wat is de prijs van het uitje?",
            },
            Object {
              "content": "Daarom is de prijs niet zomaar te noemen zonder uw informatie. Alle aanbieders maken voor u graag een offerte op maat. Indien u het aanvraagformulier onder het gekozen uitje invult, met uw wensen en aanvullingen, maken wij een offerte op maat voor u. Na het invullen ontvangt u ook direct het telefoonnummer van de organisator, zodat u indien gewenst ook direct telefonisch contact kunt opnemen.",
              "title": "Telefonisch contact",
            },
            Object {
              "content": "Daarom is de prijs niet zomaar te noemen zonder uw informatie. Alle aanbieders maken voor u graag een offerte op maat. Indien u het aanvraagformulier onder het gekozen uitje invult, met uw wensen en aanvullingen, maken wij een offerte op maat voor u. Na het invullen ontvangt u ook direct het telefoonnummer van de organisator, zodat u indien gewenst ook direct telefonisch contact kunt opnemen.",
              "title": "Kan WegmetdeBaas mij helpen een bedrijfsuitje samen te stellen?",
            },
          ]
        }
        modify="show-desktop"
      />
    </div>
    <div>
      <h2>
        Andere vragen?
      </h2>
      <p>
        Mail ons het aanvraagformulier of direct via info@wegmetdebaas.nl
      </p>
      <WithFormik(InnerForm)
        onSubmit={[Function]}
      />
    </div>
  </div>
</Section>
`;
